primitives/io/src/lib.rs

```diff
@@ -1239,6 +1246,49 @@ pub trait Sandbox {
 	}
 }
 
+/// Runtime spawn extension.
+pub trait RuntimeSpawn : Send {
+	/// Create new runtime instance and use dynamic dispatch to invoke with specified payload.
+	///
+	/// Returns handle of the spawned task.
+	///
+	/// Function pointers (`dispatcher_ref`, `func`) are WASM pointer types.
+	fn spawn_call(&self, dispatcher_ref: u32, func: u32, payload: Vec<u8>) -> u64;
+
+	/// Join the result of previously created runtime instance invocation.
+	fn join(&self, handle: u64) -> Vec<u8>;
+}
+
+#[cfg(feature = "std")]
+sp_externalities::decl_extension! {
+	/// Extension that supports spawning extra runtime instances in externalities.
+	pub struct RuntimeSpawnExt(Box<dyn RuntimeSpawn>);
+}
+
+/// Wasm host functions for managing tasks.
+///
+/// This should not be used directly. Use `sp_io::tasks` for running parallel tasks instead.
+#[runtime_interface]
```

Comment:
```suggestion
#[runtime_interface(wasm_only)]
```

To make sure we don't call this from native ;)

Reply:
wasm-only interface can't access externalities and extensions, can it?

Reply:
It is not implemented for externalities, but you can just access them.

I once thought about implementing it for both, but there wasn't the need until now.